### PR TITLE
Ports carrier visible huggers from CM part 2

### PR DESCRIPTION
**Pull request by Dgeen**
Ports carrier visible huggers from CM

## About The Pull Request

With Blundir's permission, I'm reopening his pr #11735 
Code (and only it) has been changed, I hope its quality will suit Tivi

## Why It's Good For The Game

More visual depth and detail, general visual improvement that will help visually identify amount of huggers the carrier has, good for both xenos and marines.

## Changelog

:cl:
add: ported carrier hugger overlay
imageadd: added hugger images for carrier
/:cl:

### CODE REVIEW

**Comment by Coastfront** [comment]
I always knew the pregnancy xeno was missing "getting naked before doing the dirty" and yet, just as I couldn't have known my favorite food without first seeing it, I had no idea that I even knew this was missing from my life.

Thank you.

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 1]
```diff
@@ -17,12 +17,49 @@
...
+/mob/living/carbon/xenomorph/carrier/Stat()
+	. = ..()
 	if(statpanel("Game"))
 		stat("Stored Huggers:", "[huggers] / [xeno_caste.huggers_max]")
+
+/mob/living/carbon/xenomorph/carrier/update_icons()
+	. = ..()
+	update_hugger_overlays()
+
+/mob/living/carbon/xenomorph/carrier/proc/update_hugger_overlays()
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:39]: Why are you making a separate proc? Just use the existing update_overlays proc.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm (hunk 1 above)
Comment (Dgeen) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm]: Done?

Comment (Lumipharon) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm]: Update_icons already calls update_overlays by default, so you don't have to specifically call it. This is part of why its important to use that proc as it's already built in.

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm [hunk 2]
```diff
@@ -17,12 +17,49 @@
...
+	. = ..()
 	if(statpanel("Game"))
 		stat("Stored Huggers:", "[huggers] / [xeno_caste.huggers_max]")
+
+/mob/living/carbon/xenomorph/carrier/update_icons()
+	. = ..()
+	update_overlays()
+
+/mob/living/carbon/xenomorph/carrier/update_overlays()
+	. = ..()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm:40]:
```suggestion
```
update_overlays is part of the standard icon update procs and shouldnt be called directly, you can just put this in normally

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest is good enough

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm (hunk 2 above)
Comment (Dgeen) [reply code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm]: Done

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**